libsupport/src/ArrowInterchange.cpp

```diff
@@ -48,19 +49,51 @@ ApproxArrayDataMemUse(const std::shared_ptr<arrow::ArrayData>& data) {
 
 }  // anonymous namespace
 
-std::shared_ptr<arrow::ChunkedArray>
+katana::Result<std::shared_ptr<arrow::Table>>
+katana::TakeRows(
+    const std::shared_ptr<arrow::Table>& original,
+    const std::vector<bool>& picker) {
```

Comment:
If you make an arrow::BooleanArray instead of vector<bool> and perform the operation column-wise, this would match the [Filter](https://arrow.apache.org/docs/cpp/compute.html#selections) API. That should be meaningfully more efficient than ConcatenateTables

Reply:
Yeah, I like this idea.  I'll look into it.